libs/aws/langchain_aws/agents/utils.py

```diff
@@ -0,0 +1,287 @@
+from __future__ import annotations
+
+import json
+import logging
+import time
+from collections import defaultdict
+from typing import Any, List, Optional, Tuple, Union
+
+import boto3
+from botocore.client import Config
+from langchain_core.tools import BaseTool
+
+from langchain_aws.agents.types import (
+    BedrockAgentAction,
+    BedrockAgentFinish,
+    GuardrailConfiguration,
+    OutputType,
+    _DEFAULT_ACTION_GROUP_NAME,
+)
+
+
+def get_boto_session(
+    credentials_profile_name: Optional[str] = None,
+    region_name: Optional[str] = None,
+    endpoint_url: Optional[str] = None,
+) -> Any:
+    """
+    Construct the boto3 session
+    """
+    if credentials_profile_name:
+        session = boto3.Session(profile_name=credentials_profile_name)
+    else:
+        # use default credentials
+        session = boto3.Session()
+    client_params = {
+        "config": Config(
+            connect_timeout=120, read_timeout=120, retries={"max_attempts": 0}
+        )
+    }
+    if region_name:
+        client_params["region_name"] = region_name
+    if endpoint_url:
+        client_params["endpoint_url"] = endpoint_url
+    return client_params, session
+
+
+def parse_agent_response(response: Any) -> OutputType:
+    """
+    Parses the raw response from Bedrock Agent
+
+    Args:
+        response: The raw response from Bedrock Agent
+
+    Returns
+        Either a BedrockAgentAction or a BedrockAgentFinish
+    """
+    response_text = ""
+    event_stream = response["completion"]
+    session_id = response["sessionId"]
+    trace_log_elements = []
+    files = []
+    for event in event_stream:
```

Comment:
Should we have some sort of signal for unknown event types so consumers can more easily identify service model changes?

Reply:
Good point, will update this separately.

Reply:
Currently this is not a type within the response, will investigate more and update this in a later PR